auction-app-backend/src/main/java/com/atlantbh/auctionappbackend/filter/CustomAuthenticationFilter.java

```diff
@@ -0,0 +1,76 @@
+package com.atlantbh.auctionappbackend.filter;
+
+import com.atlantbh.auctionappbackend.security.JwtConfig;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+public class CustomAuthenticationFilter extends UsernamePasswordAuthenticationFilter {
```

Comment:
Not sure how we plan to use this and `CustomAuthorizationFilter` classes. We can discuss this. Also, I suggest adding Javadoc to make it easier to understand things like this.